tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/PopulateBasicPropertiesHeadersIntegration.cs

```diff
@@ -45,7 +51,29 @@ public class PopulateBasicPropertiesHeadersIntegration
             return new CallTargetReturn<TReturn?>(returnValue);
         }
 
-        returnValue ??= CachedBasicPropertiesHelper<TReturn>.CreateHeaders();
+        // PopulateBasicPropertiesHeaders returns null if the supplied IReadOnlyBasicProperties
+        // does not have to be modified or if it's a writable instance.
+        // If that is the case then we have to fetch IReadOnlyBasicProperties from the argument
+        // list instead of creating a new instance that overwrites the supplied properties.
+        if (returnValue is null)
+        {
+            if (state.State.DuckIs<IBasicProperties>())
```

Comment:
I don't think this will _ever_ pass currently 🤔 `state.State` is _not_ the original type, it's the (boxed) struct duck type _around_ `IReadOnlyBasicProperties`...

I think I would stop doing the duck typing in the `OnMethodBegin`, and do it here, using `TryDuckCast` and then passing in the original object (not the duck-typed object) into CreateHeaders?

Reply:
I removed the generic type constraints from `OnMethodBegin()` so we no longer duck-type the original `IReadOnlyBasicProperties` which we pass as the state.